From 678365fa1927880cad2e864c9a1ff7597b2cd767 Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Wed, 20 Aug 2025 06:23:06 +0200
Subject: [PATCH] libfs_avb: Allow LKs patched with fenrir to boot on AOSP

fenrir spoof bootloader state to green, however in doing so it result in AVB failures.

Ported from Nothing-2A/android_system_core commit 8ff6e7a68523c3b870d8dcd5713c71ea15b43dd2. Android 17 moved fs_mgr into system/fs/fs_mgr; apply the same change in its new repository.

Change-Id: I75b28942684c736492ee2a605f3f137dd0d89b2e
---
 libfs_avb/util.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfs_avb/util.cpp b/libfs_avb/util.cpp
index 79c7b949..f78e89cd 100644
--- a/libfs_avb/util.cpp
+++ b/libfs_avb/util.cpp
@@ -111,7 +111,7 @@ bool IsDeviceUnlocked() {
     std::string verified_boot_state;
 
     if (fs_mgr_get_boot_config("verifiedbootstate", &verified_boot_state)) {
-        return verified_boot_state == "orange";
+        return (verified_boot_state == "orange" || verified_boot_state == "green");
     }
     return false;
 }
